### PR TITLE
feat: alarms for ses and spf records as default

### PR DESCRIPTION
**Pull request by evan-boone**
- SPF-record opprettes automatisk med default-verdi (som kan overstyres)
- SES-eventer logges til CloudWatch
- Varsling på høy bounce og complain rate

### CODE REVIEW

File: src/ses/sesdomain/index.ts [hunk 1]
```diff
@@ -51,6 +57,9 @@ export class SesDomain extends constructs.Construct {
         DefaultConfigurationSetName: props.defaultConfigurationSetName,
         // Bump this if changing logic in the lambda that should be
```
Comment (aleksil) [inline src/ses/sesdomain/index.ts:58]: Kommentaren her ligger nå på feil felt

File: src/ses/sesdomain/index.ts [hunk 2]
```diff
@@ -32,6 +32,12 @@ interface Props {
    * Default configuration set for emails sent from this domain.
    */
   defaultConfigurationSetName?: string
+  /**
+   * Values to be included in SPF TXT record
+   *
+   * @default ["v=spf1 include:amazonses.com ~all"]
+   */
+  spfRecordValues?: string[]
```
Comment (aleksil) [inline src/ses/sesdomain/index.ts:40]: Dette git ikke mening. Man kan ikke ha flere SPF TXT-records for et gitt domene, og om disse blir konkatenert til en record så vil alt etter `-all` feile uansett (ref eksempelet du har i snapshots).

Jeg tror det gir mer mening å kun ta inn en enkelt string, siden vi uansett kun skal produsere en record.

Comment (aleksil) [reply src/ses/sesdomain/index.ts:40]: Dette blir forøvrig en breaking change, siden vi i de fleste løsningene manuelt legger til SPF-records, så disse må fjernes samtidig som man bumper til ny versjon av liflig-cdk der dette legges til av oss.

**Review by aleksil — CHANGES_REQUESTED** [review]

File: src/ses/sesdomain/index.ts [hunk 3]
```diff
@@ -32,6 +32,12 @@ interface Props {
    * Default configuration set for emails sent from this domain.
    */
   defaultConfigurationSetName?: string
+  /**
+   * Values to be included in SPF TXT record
+   *
+   * @default ["v=spf1 include:amazonses.com ~all"]
```
Comment (aleksil) [inline src/ses/sesdomain/index.ts:38]: Skal ikke lenger være en array her

File: src/ses/sesdomain/index.ts [hunk 4]
```diff
@@ -32,6 +32,12 @@ interface Props {
...
+   */
+  spfRecordValues?: string
```
Comment (aleksil) [inline src/ses/sesdomain/index.ts:40]: Kan med fordel droppe flertall og endre til `spfRecordValue` her og andre steder

**Review by aleksil — APPROVED** [review]
La til to nits, ellers ser det bra ut nå :+1:

File: src/alarms/ses-alarms.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+  constructor(scope: constructs.Construct, id: string, props: SesAlarmsProps) {
+    super(scope, id)
+
+    this.action = props.action
+
+    const bouncedMessagesAlarm = new cloudwatch.Metric({
+      metricName: "Reputation.BounceRate",
+      namespace: "AWS/SES",
+      statistic: "Maximum",
+      unit: Unit.PERCENT,
```
Comment (stekern) [inline src/alarms/ses-alarms.ts:69]: I'm not sure if this actually works as expected. As far as I know, this metric does not actually have any unit set for its datapoints. I think you may want to remove the unit altogether here, and update the threshold value accordingly

File: src/alarms/ses-alarms.ts [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+      period: cdk.Duration.days(1),
+    }).createAlarm(this, "ComplaintMessagesAlarm", {
+      alarmDescription: `The SES complaint rate is over ${
+        props?.complaintRateAlarm?.threshold ?? 0.07
+      }/%`,
+      comparisonOperator: cloudwatch.ComparisonOperator.GREATER_THAN_THRESHOLD,
+      evaluationPeriods: 1,
+      threshold: props?.complaintRateAlarm?.threshold ?? 0.07,
```
Comment (stekern) [inline src/alarms/ses-alarms.ts:99]: I think you'll need to divide the threshold value by 100. May also want to consider just making the threshold half of what it takes to get an account under review (e.g., `0.0005`)

File: src/alarms/ses-alarms.ts [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+      bouncedMessagesAlarm.addAlarmAction(
+        props?.bouncedMessagesAlarm?.action || this.action,
+      )
+    }
+
+    const complaintMessagesAlarm = new cloudwatch.Metric({
+      metricName: "Reputation.ComplaintRate",
+      namespace: "AWS/SES",
+      statistic: "Maximum",
+      unit: Unit.PERCENT,
```
Comment (stekern) [inline src/alarms/ses-alarms.ts:91]: I'm not sure if this actually works as expected. As far as I know, this metric does not actually have any unit set for its datapoints. I think you may want to remove the unit altogether here, and update the threshold value accordingly

File: src/alarms/ses-alarms.ts [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+    super(scope, id)
+
+    this.action = props.action
+
+    const bouncedMessagesAlarm = new cloudwatch.Metric({
+      metricName: "Reputation.BounceRate",
+      namespace: "AWS/SES",
+      statistic: "Maximum",
+      unit: Unit.PERCENT,
+      period: cdk.Duration.days(1),
```
[the remark this answers is not in the document]
Comment (evan-boone) [reply src/alarms/ses-alarms.ts:70]: If the period is so short, won't an alarm most likely get sent almost every time a BOUNCE or COMPLAINT happens?

File: src/ses/configurationsetsnsdestination/index.ts [hunk 9]
```diff
@@ -22,6 +22,11 @@ export type ConfigurationSetSnsDestinationEventType =
   | "SUBSCRIPTION"
 
 export interface ConfigurationSetSnsDestinationProps {
+  /**
+   * Whether SES events will be logged to SNS
```
Comment (stekern) [inline src/ses/configurationsetsnsdestination/index.ts:26]: nit: I think mentioning CloudWatch here would be more accurate and descriptive, as that is what this toggle effectively does. `Whether SES events should be logged to CloudWatch`?

**Review by stekern — APPROVED** [review]
LGTM 🚀 

(You may want to do a rebase to squash the commits and get a cleaner commit history, or just a squash and merge through the GitHub UI)

**Pull request merged**